streams/src/main/java/org/apache/kafka/streams/kstream/internals/KGroupedStreamImpl.java

```diff
@@ -391,20 +390,21 @@ public V apply(final K aggKey, final V aggOne, final V aggTwo) {
             final String functionName,
             final StateStoreSupplier storeSupplier) {
 
-        final String aggFunctionName = topology.newName(functionName);
+        final String aggFunctionName = builder.newName(functionName);
 
         final String sourceName = repartitionIfRequired(storeSupplier.name());
 
-        topology.addProcessor(aggFunctionName, aggregateSupplier, sourceName);
-        topology.addStateStore(storeSupplier, aggFunctionName);
-
-        return new KTableImpl<>(topology,
-                aggFunctionName,
-                aggregateSupplier,
-                sourceName.equals(this.name) ? sourceNodes
-                        : Collections.singleton(sourceName),
-                storeSupplier.name(),
-                isQueryable);
+        builder.internalTopologyBuilder.addProcessor(aggFunctionName, aggregateSupplier, sourceName);
```

Comment:
I don't think, the code would be simpler with the suggested change. ATM, the code has nested "hierarchy" with (top level to inner) `Topology -> InternalTopologyBuilder` -- thus, `InternalTopologyBuilder` does not even know it's "wrapped" by `Topology`.

Also, IMHO, in internal classes, it's ok to use `InternalTopologyBuilder`.